playground/cdk/CdkSample.AppHost/mykv.module.bicep

```diff
@@ -1,51 +1,32 @@
-targetScope = 'resourceGroup'
-
-@description('')
-param location string = resourceGroup().location
-
-@description('')
-param principalId string
-
-@description('')
-param principalType string
-
-@description('')
-param signaturesecret string
+resource mykv 'Microsoft.KeyVault/vaults@2019-09-01' = {
+    name: take('mykv-${uniqueString(resourceGroup().id)}', 24)
+    location: resourceGroup().location
+    properties: { }
+    tags: {
+        'aspire-resource-name': 'mykv'
+    }
+}
 
+output vaultUri string = mykv.properties.vaultUri
 
-resource keyVault_aMZbuK3Sy 'Microsoft.KeyVault/vaults@2022-07-01' = {
-  name: toLower(take('mykv${uniqueString(resourceGroup().id)}', 24))
-  location: location
-  tags: {
-    'aspire-resource-name': 'mykv'
-  }
-  properties: {
-    tenantId: tenant().tenantId
-    sku: {
-      family: 'A'
-      name: 'standard'
+resource KeyVaultAdministrator_mykv 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+    name: guid(resourceGroup().id, 'KeyVaultAdministrator_mykv')
+    properties: {
+        principalId: principalId
+        roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '00482a5a-887f-4fb3-b363-3b7fe8e74483')
+        principalType: principalType
     }
-    enableRbacAuthorization: true
-  }
+    scope: mykv
 }
 
-resource roleAssignment_hVU9zjQV1 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
-  scope: keyVault_aMZbuK3Sy
-  name: guid(keyVault_aMZbuK3Sy.id, principalId, subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '00482a5a-887f-4fb3-b363-3b7fe8e74483'))
-  properties: {
-    roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '00482a5a-887f-4fb3-b363-3b7fe8e74483')
-    principalId: principalId
-    principalType: principalType
-  }
+resource mysecret 'Microsoft.KeyVault/vaults/secrets@2019-09-01' = {
+    name: take('mysecret-${uniqueString(resourceGroup().id)}', 127)
+    properties: {
+        value: signaturesecret
+    }
+    parent: mykv
 }
 
-resource keyVaultSecret_mW5tlkNij 'Microsoft.KeyVault/vaults/secrets@2022-07-01' = {
-  parent: keyVault_aMZbuK3Sy
-  name: 'mysecret'
-  location: location
-  properties: {
-    value: signaturesecret
-  }
-}
+param principalId string
```

Comment:
The location parameter is missing, we need to add that everywhere.

Reply:
It's a known issue.  You're getting your values from https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/provisioning/Azure.Provisioning/src/Primitives/LocationPropertyResolver.cs and we need to update that to inject/reuse a parameter if available.  You can explicitly work around this yourself by setting `Location` or wait for it to be fixed in Azure.Provisioning.

Reply:
Switched to using a param in https://github.com/Azure/azure-sdk-for-net/pull/45911